src/JTracker/View/Renderer/TrackerExtension.php

```diff
@@ -202,6 +226,226 @@ public function fetchAvatar($userName = '', $width = 0, $class = '')
+				[],
+				2
+			);
+
+			if ($response->code !== 200)
+			{
+				return g11n3t('Could not load template section.');
+			}
+
+			$body = $response->body;
+
+			// Replace the common placeholders
+			$body = strtr(
+				$body,
+				[
+					'%reportroute%'  => $this->app->get('uri.base.path') . 'tracker/jtracker/add',
+					'%currentyear%' => date('Y'),
+				]
+			);
+
+			// Replace the context aware placeholders
+			if ($this->app->getUser()->id)
+			{
+				$body = strtr(
+					$body,
+					[
+						'%loginroute%' => $this->app->get('uri.base.path') . 'logout',
+						'%logintext%'  => g11n3t('Log out'),
+					]
+				);
+			}
+			else
+			{
+				$body = strtr(
+					$body,
+					[
+						'%loginroute%' => $this->loginHelper->getLoginUri(),
+						'%logintext%'  => g11n3t('Log in'),
+					]
+				);
+			}
+
+			return $body;
+		};
+
+		if ($this->app->get('cache.enabled', false))
+		{
+			/** @var \Psr\Cache\CacheItemPoolInterface $cache */
+			$cache = $container->get('cache');
+
+			if ($cache->hasItem($key))
+			{
+				$item = $cache->getItem($key);
+
+				// Make sure we got a hit on the item, otherwise we'll have to re-cache
+				if ($item->isHit())
+				{
+					$body = $item->get();
+				}
+				else
+				{
+					try
+					{
+						$body = $fetchPage();
+
+						$item = (new Item($key, $this->app->get('cache.lifetime', 900)))
+							->set($body);
+
+						$cache->save($item);
+					}
+					catch (\RuntimeException $e)
+					{
+						$body = 'Could not load template section.';
+					}
+				}
+			}
+			else
+			{
+				try
+				{
+					$body = $fetchPage();
+
+					$item = (new Item($key, $container->get('app')->get('cache.lifetime', 900)))
+						->set($body);
+
+					$cache->save($item);
+				}
+				catch (\RuntimeException $e)
+				{
+					$body = 'Could not load template section.';
+				}
+			}
+		}
+		else
+		{
+			try
+			{
+				$body = $fetchPage();
+			}
+			catch (\RuntimeException $e)
+			{
+				$body = 'Could not load template section.';
+			}
+		}
+
+		return $body;
+	}
+
+	/**
+	 * Fetches and renders the CDN menu element, optionally caching the data.
+	 *
+	 * @return  string
+	 *
+	 * @since   1.0
+	 */
+	public function getCdnMenu()
+	{
+		$key = md5(get_class($this) . '::' . __METHOD__ . ' - language - ' . $this->app->getLanguageTag());
+
+		$fetchPage = function ()
+		{
+			// Set a very short timeout to try and not bring the site down
+			$response = HttpFactory::getHttp()->get(
+				'https://cdn.joomla.org/template/renderer.php?section=menu&language=' . $this->app->getLanguageTag(),
+				[],
+				2
+			);
+
+			if ($response->code !== 200)
+			{
+				return g11n3t('Could not load template section.');
+			}
+
+			$body = $response->body;
+
+			// Remove the search module
+			$body = str_replace("\t<div id=\"nav-search\" class=\"navbar-search pull-right\">\n\t\t<jdoc:include type=\"modules\" name=\"position-0\" style=\"none\" />\n\t</div>\n", '', $body);
```

Comment:
FILE: ...is/build/joomla/jissues/src/JTracker/View/Renderer/TrackerExtension.php
--------------------------------------------------------------------------------
FOUND 0 ERROR(S) AND 1 WARNING(S) AFFECTING 1 LINE(S)
--------------------------------------------------------------------------------
 380 | WARNING | Line exceeds 150 characters; contains 184 characters
--------------------------------------------------------------------------------
UPGRADE TO PHP_CODESNIFFER 2.0 TO FIX ERRORS AUTOMATICALLY
--------------------------------------------------------------------------------